### PR TITLE
Add note about using dag_run.conf in BashOperator

### PR DESCRIPTION
**Pull request by ashb**
---
Make sure to mark the boxes below before creating PR: [x]

- [x] Description above provides context of the change
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Target Github ISSUE in description if exists
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).

---
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by kaxil — APPROVED** [review]

**Review by kaxil — COMMENTED** [review]
Doc building error

```
==================== Error   2 ====================
 WARNING: Could not lex literal_block as "python". Highlighting skipped.

File path: /opt/airflow/docs/howto/operator/bash.rst (67)

  63 | 
  64 | Instead, you should pass this via the ``env`` kwarg and use double-quotes
  65 | inside the bash_command, as below:
  66 | 
  67 | .. code-block:: python
  68 | 
  69 |     bash_task = BashOperator(
  70 |         task_id="bash_task",
  71 |         bash_command='echo "here is the message: \\'$message\\'"',
  72 |         env={'message': '{{ dag_run.conf["message"] if dag_run else "" }}'},
==================================================
```

**Comment by ashb** [comment]
Hmmmm, I wonder what is going on between sphinx and pygmentize:

```
airflow ❯ pygmentize -l python -F raiseonerror < test-lex.py
    bash_task = BashOperator(
        task_id="bash_task",
        bash_command='echo "here is the message: \'$message\'"',
        env={'message': '{{ dag_run.conf["message"] if dag_run else "" }}'},
```

works fine and doesn't error.

**Comment by ashb** [comment]
(I tried with `\` first, that failed, and with `\\` too wondering if sphinx was treating `\\` as something. That failed too)

File: docs/howto/operator/bash.rst [hunk 1]
```diff
@@ -41,6 +41,37 @@ You can use :ref:`Jinja templates <jinja-templating>` to parameterize the
...
+    )
+
+Instead, you should pass this via the ``env`` kwarg and use double-quotes
+inside the bash_command, as below:
+
+.. code-block:: python
+
+    bash_task = BashOperator(
+        task_id="bash_task",
+        bash_command='echo "here is the message: \'$message\'"',
```
Comment (kaxil) [inline docs/howto/operator/bash.rst:71]: This lin failed again, I mean you can remove the quotes after echo as it is just bash.

Example:

```suggestion
        bash_command='echo here is the message: "$message"',
```

Comment (ashb) [reply docs/howto/operator/bash.rst:71]: I guess I can. The point was to try and get output when run like

> here is the message 'foo bar'

Comment (ashb) [reply docs/howto/operator/bash.rst:71]: Running it locally, the problem does appear to be around `\`

image

Comment (ashb) [reply docs/howto/operator/bash.rst:71]: Oh pay attention ash. It's not this file.

Comment (ashb) [reply docs/howto/operator/bash.rst:71]: https://github.com/apache/airflow/pull/9143/commits/c8fd7de369dff5fcf9b5950b034330292f78062b

**Review by kaxil — APPROVED** [review]

**Pull request merged**